### PR TITLE
Remove keyring path before signing with new key; Remove message on no keyring

### PR DESCRIPTION
**Pull request by Bastian-Krause**
`sign_bundle()` verifies the signature after signing against the given keyring. When resigning this keyring is most likely not valid anymore. So remove it and skip signature verification after signing.

Invalidating the keyring leads to the error message being printed although the user gave the required `--keyring` option.

As the message has not much benefit anyway simply remove it altogether.

Fixes: #154, ca64249 ("src/bundle: always call check_bundle explicitly")

Note: this has nothing to do with #460.

### CODE REVIEW

File: src/bundle.c [hunk 1]
```diff
@@ -474,6 +474,9 @@ gboolean resign_bundle(RaucBundle *bundle, const gchar *outpath, GError **error)
 		goto out;
 	}
 
+	/* invalidate old keyring */
```
Comment (ejoerns) [inline src/bundle.c]: This does not really reflect why this is needed. Maybe


```suggestion
	/* invalidate to not trigger post-signing verification with inappropriate keyring */
```

Comment (Bastian-Krause) [reply src/bundle.c]: "invalidate keyring to avoid post-signing verification with inappropriate keyring"?

Comment (ejoerns) [reply src/bundle.c]: would accept it ;)

Comment (Bastian-Krause) [reply src/bundle.c]: Done.

**Comment by ejoerns** [comment]
@Bastian-Krause I've sponsored some test adaptions ;)

**Comment by Bastian-Krause** [comment]
> @Bastian-Krause I've sponsored some test adaptions ;)

Thanks!

**Comment by ejoerns** [comment]
I've squashed it and rebased it onto latest master.

File: test/bundle.c [hunk 2]
```diff
@@ -172,7 +172,7 @@ static void bundle_test_resign(BundleFixture *fixture,
 	 * Note we have to use r_context() here as a hack to avoid re-setting
 	 * the context's 'pending' flag which would cause a re-initialization
 	 * of context and thus overwrite content of 'config' member. */
-	r_context()->config->keyring_path = g_strdup("test/openssl-ca/rel-ca.pem");
+	r_context_conf()->keyringpath = g_strdup("test/openssl-ca/rel-ca.pem");
```
Comment (jluebbe) [inline test/bundle.c]: Why did you change r_context to r_context_conf? Doesn't the comment apply any more?

**Comment by jluebbe** [comment]
Closing in favor of #498.
